docs/install.rst

```diff
@@ -35,6 +35,22 @@ following line in your ``.zshrc`` after activating Antigen::
 
    antigen bundle nojhan/liquidprompt
 
+Installation via Zinit
+----------------------
+
+To install via `Zinit <https://github.com/zdharma/zinit>` _, simply add the following line in your ``.zshrc`` after activating Zinit::
+
+   zinit load nojhan/liquidprompt
+
+Zinit provides an additional `Ice Modifier <http://zdharma.org/zinit/wiki/INTRODUCTION/#some_ice-modifiers>` _ syntax, which can be used to improve the startup time of your terminal::
+
+   zinit ice load'[[ $- = *i* ]]' \
+             wait'!' \
+             lucid \
+             nocd \
+             atload'!__lp_set_prompt'
```

Comment:
I haven't tested this. I'm assuming that this does work? I see two possibilities:
1. The `light` command makes the file load, but not execute. Hence the atload part, telling it what command to run in the prompt.
2. The atload part is duplicating definition, so _lp_set_prompt is run twice. Liquidprompt does after all setup Zsh all by itself correctly when not used in a plug-in manager.

I'm guessing it is the latter. I'll need to do a bit of testing to dissuade my worries. I don't want to add something to the docs that isn't completely correct. I'll try to do some testing later tonight.

Reply:
> The light command makes the file load, but not execute. Hence the atload part, telling it what command to run in the prompt.

`light` doesn't change whether the file loads or not, it just changes what `zinit` does while it's loading, that is to say: it doesn't provide information in the `zinit report` command, which is useful for profiling.  [You can read about that here.](https://zdharma.org/zinit/wiki/INTRODUCTION/#basic_plugin_loading).

> The atload part is duplicating definition, so _lp_set_prompt is run twice. Liquidprompt does after all setup Zsh all by itself correctly when not used in a plug-in manager.

The `ice` part of the snippet is telling `zinit` to defer execution, and when it's ready to execute, the `__lp_set_prompt` function is evaluated, so I don't believe it is run twice. The advantage of doing things this was is that the terminal becomes usable immediately - even before liquidprompt is ready.

Reply:
The issue is that when the Liquidprompt file is sourced, it configures Zsh to run that function. So if you also tell zinit to do that also, will it run twice?

Reply:
@Rycieos I'm not sure how definitive this is, but I added an echo to the first line of `__lp_set_prompt` and then started a new terminal, and the echo only appears once. Is that helpful?

Reply:
Hi again @Rycieos,

> It is not faster. Doing only zinit light nojhan/liquidprompt is the same speed, the lazy loading doesn't help (at least on my machine).

The performance improvement comes as a result of its asynchronous loading - if you have liquidprompt in combination with other plugins which all would otherwise load at startup, the cumulative effect of using zinit provides a noticeable performance improvement (anecdotally!)

> Because it is lazy, I can see it flash for a second as the default prompt displays then is replaced.

Yes, this is absolutely a valid point

> It is way more configuration. Why do we need to recommend all that when the simple command works just as well?

My intention was to produce the most idiomatically correct implementation of the config (it's the one I use myself) to make the most of zinit's featureset. If someone truly wants to use zinit in the way I am, I guess we've probably produced enough of a papertrail with this PR that they'll find what they're looking for - even if it's not in the docs, so on that level I don't really mind much :)

> Loading this way, Liquidprompt displays an exit code 1 on the first prompt. With only a load or light command, it does not.

Huh, that *is* a good point - and not something I noticed myself. I wonder why...

Yep, I think you make a salient case for dropping it, and I agree that the last one in particular is a showstopper. I'll remove it in my next commit!

Reply:
Belay that order!

The `ver` command does not work as I described it. I just tested it, it seems that it needs to be part of an `ice` command? So unless you can figure out the correct way to use it as part of a `load` or `light`, I think we need an `ice` to target the stable branch.

Reply:
Aye aye cap'n :smile: 

Yes you are right, it was my intention to drop the lazy/async aspects of the ice modifier stuff only!

Reply:
Just realized I forgot the other reason not to do that: Telling users to reference `__lp_set_prompt()` is not a good idea; it's a private function. While there are no plans to change that specific part of the internal API, it is a part that is not guaranteed to not change between versions, and I don't want users depending on that.

Obviously I can't stop power users like yourself from making it work to get lazy loading working, but at least I can prevent the average user from doing that.

Though maybe this should be a place for enhancement. It would be nice for Liquidprompt to support lazy/async loading, at least for Zsh. I would probably need some help though.